File: Selenium/Activities/Activity11.java
package activity;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity11 {

	public static void main(String[] args) {
		
		WebDriver driver = new ChromeDriver();
		WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(2000));
		
		driver.get("https://training-support.net/webelements/dynamic-controls");
		
		System.out.println("Title of the page: " + driver.getTitle());
		
		WebElement checkbox = driver.findElement(By.cssSelector("#checkbox"));
		System.out.println("Is checkbox present? "+ checkbox.isDisplayed());
		
		WebElement toggleBtn = driver.findElement(By.xpath("//button[text()='Toggle Checkbox']"));
		toggleBtn.click();
		
		wait.until(ExpectedConditions.invisibilityOf(checkbox));
		System.out.println("Is checkbox present after toggling off? "+ checkbox.isDisplayed());
		
		toggleBtn.click();
		wait.until(ExpectedConditions.elementToBeClickable(checkbox));
		checkbox.click();
		
		System.out.println("Is checkbox selected? "+ checkbox.isSelected());
		
		driver.quit();

	}

}

File: Selenium/Activities/activity_11.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.wait import WebDriverWait

with webdriver.Firefox() as driver:
    wait = WebDriverWait(driver, timeout=5)

    driver.get("https://training-support.net/webelements/dynamic-controls")

    print("Title of the page: ", driver.title)

    checkbox = driver.find_element(By.CSS_SELECTOR, "#checkbox")
    print("Is checkbox present? ", checkbox.is_displayed())

    toggle_btn = driver.find_element(By.XPATH, "//button[text()='Toggle Checkbox']")
    toggle_btn.click()

    wait.until(EC.invisibility_of_element(checkbox))
    print("Is checkbox present after toggling off? ", checkbox.is_displayed())

    toggle_btn.click()
    wait.until(EC.element_to_be_clickable(checkbox))
    checkbox.click()

    print("Is checkbox selected after clicking? ", checkbox.is_selected())

    driver.quit()